metadata tests fail with tskit 3.6dev1

With `pip install tskit` installing tskit0.3.5, all tests pass, but if I do `pip install git+https://github.com/tskit-dev/tskit#subdirectory=python`, which installs `tskit 3.6dev1`, a few metadata tests fail, e.g. 

```
**___________________________________________ TestMetadataRoundTrip.test_individual_metadata_subset[True] _____________________________________________
...
>       assert all_metadata == output_metadata
E       AssertionError: assert [{'0': 'Ad', ...G', ...}, ...] == [{'0': 'Ad', ...G', ...}, ...]
E         At index 0 diff: {'0': 'Ad', '1': 'ct', '2': 'X', '3': 'bpCqmZCk'} != {'sample_data_time': 0, '0': 'Ad', '1': 'ct', '2': 'X', '3': 'bpCqmZCk'}
E         Use -v to get the full diff

tests/test_inference.py:982: AssertionError
...

FAILED tests/test_inference.py::TestMetadataRoundTrip::test_individual_metadata_subset[True] - AssertionError: assert [{'0': 'Ad', ...G', ...}, ......
FAILED tests/test_inference.py::TestMetadataRoundTrip::test_individual_metadata[True] - AssertionError: assert [{}, {'0': 't...UFkIVt'}, ...] == [{...

```

--- a/tests/test_inference.py
+++ b/tests/test_inference.py
@@ -948,7 +948,10 @@ class TestMetadataRoundTrip:
             individual.metadata if use_schema else json.loads(individual.metadata)
             for individual in output_ts.individuals()
         ]
-        assert all_metadata == output_metadata
+        # output metadata can have some extra fields, e.g. "sample_data_time"
+        # so check all_metadata is "contained in" output_metadata
+        for all, output in zip(all_metadata, output_metadata):
+            assert all.items() <= output.items()
 
     @pytest.mark.parametrize("use_schema", [True, False])
     def test_individual_metadata_subset(self, use_schema):
@@ -979,9 +982,12 @@ class TestMetadataRoundTrip:
             individual.metadata if use_schema else json.loads(individual.metadata)
             for individual in output_ts.individuals()
         ]
-        assert all_metadata == output_metadata
+        # output metadata can have some extra fields, e.g. "sample_data_time"
+        # so check all_metadata is "contained in" output_metadata
+        for all, output in zip(all_metadata, output_metadata):
+            assert all.items() <= output.items()
         for j, metadata in enumerate(sample_data.individuals_metadata[:]):
-            assert all_metadata[j] == metadata
+            assert all_metadata[j].items() <= metadata.items()
 
         # Now do this for various subsets of the data and make sure
         # that metadata comes through correctly.

